### PR TITLE
feat: delete conversations events navigation to home on deleted (AR-2041)

### PR DESCRIPTION
**Pull request by yamilmedina**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/AR-2041" title="AR-2041" target="_blank">SubtaskAR-2041</a>  (ui) Navigation from conversation detail delete
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

Given that we receive a conversation deleted event, we need to react (remove it from the conversations list) and if we are on the details view, we need to navigate back to the conversations lists, because this conversation is not available anymore.

### Attachments (Optional)

https://user-images.githubusercontent.com/5806454/182393652-66ca6967-81ff-4d20-a7bc-ec18a7b42a1a.mov

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [x] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [x] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [1](https://10.10.124.134/job/wire-android-reloaded/job/PR-818/1/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4048242)(4048242)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4048242)(4048242)-dev-release.apk)

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationViewModel.kt [hunk 1]
```diff
@@ -132,34 +132,43 @@ class ConversationViewModel @Inject constructor(
...
-                }
-                val conversationDetailsData = when (conversationDetails) {
-                    is ConversationDetails.Group -> ConversationDetailsData.Group(conversationDetails.conversation.id)
-                    is ConversationDetails.OneOne -> ConversationDetailsData.OneOne(conversationDetails.otherUser.id)
-                    else -> ConversationDetailsData.None
+        withContext(dispatchers.io()) {
+            observeConversationDetails(conversationId)
+                .collect { result ->
+                    when (result) {
+                        is Failure -> navigateToHome()
```
Comment (gongracr) [inline app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationViewModel.kt:139]: Niceee. I didn't know the moment a group is deleted, this use case returns a `Failure` result. That definitely makes this look way cleaner 🧹 👏

**Review by gongracr — APPROVED** [review]
Looking spotless to me! ✨ 🥇

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationViewModel.kt (hunk 1 above)
Comment (gongracr) [reply app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationViewModel.kt:139]: Small question, did you try deleting the group while the app is in airplane mode or w/o connection? would this navigate also back to home?

Comment (yamilmedina) [reply app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationViewModel.kt:139]: Yup I tried. We have to consider that this only takes care for the receiving end of a deleted convo event. 

So if we are on airplane/no connection mode, the app will do nothing, but when gaining connection, we will start processing pending events again, having this flow correctly navigated back to home.

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationViewModel.kt [hunk 2]
```diff
@@ -132,34 +132,43 @@ class ConversationViewModel @Inject constructor(
...
+                    }
                 }
-                conversationViewState = conversationViewState.copy(
-                    conversationName = conversationName,
-                    conversationAvatar = conversationAvatar,
-                    conversationDetailsData = conversationDetailsData
+        }
+    }
+
+    private fun emitConversationDetails(conversationDetails: ConversationDetails) {
```
Comment (saleniuk) [inline app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationViewModel.kt:146]: This method doesn't emit anything so I think maybe "handleConversationDetails" would be better

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationViewModel.kt (hunk 1 above)
Comment (saleniuk) [reply app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationViewModel.kt:139]: I have the same concern - we do not recognize the type of error here so will it also navigate to "home" when there is for example `NoNetworkConnection` failure? 
What about `StorageFailure`? There was a TODO about handling it and probably should be handled differently than navigating to "home" 😄

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationViewModel.kt [hunk 3]
```diff
@@ -553,6 +562,8 @@ class ConversationViewModel @Inject constructor(
         }
     }
 
+    private suspend fun navigateToHome() = navigationManager.navigate(NavigationCommand(NavigationItem.Home.getRouteWithArgs()))
```
Comment (saleniuk) [inline app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationViewModel.kt:565]: I think this command should have `BackStackMode.UPDATE_EXISTED`

**Comment by AndroidBob** [comment]
Build [6](https://10.10.124.134/job/wire-android-reloaded/job/PR-818/6/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4048777)(4048777)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4048777)(4048777)-dev-release.apk)

**Pull request merged**